packages/core/lib/commands/test/run.js

```diff
@@ -68,7 +68,42 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url exists, then it connects to that external network.
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   * If development network exists, then it connects to the development network specified in the config.
+   */
+  if (
+    config.networks[config.network] &&
+    config.network == "test" &&
+    !config.networks[config.network]["provider"] &&
+    !config.networks[config.network]["host"] &&
+    !config.networks[config.network]["url"]
```

Comment:
It would be nice to refactor for readability? Some conditional fragments may be useful further down the logic branches.
```javascript
  const configuredNetwork = config.networks[config.network];
  const testNetworkDefined = configuredNetwork && config.network == "test";
  const noProviderHostOrUrlConfigured = !configuredNetwork["provider"] && !configuredNetwork["host"] && !configuredNetwork["url"];

  if (testNetworkDefined && noProviderHostOrUrlConfigured) {
    // do the things
```

Reply:
Your suggestion in this code makes sense @cds-amal . I will make it more cleaner as suggested.